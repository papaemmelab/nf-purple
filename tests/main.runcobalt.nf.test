nextflow_process {

    name "Test Process runCobalt"
    script "main.nf"
    process "runCobalt"

    test("Should run Cobalt without failures") {

        when {
            params {
                tumor = "TEST_TUMOR"
                normal = "TEST_NORMAL"
                tumorBam = "${projectDir}/tests/data/tumor.bam"
                normalBam = "${projectDir}/tests/data/normal.bam"
                outdir = "${projectDir}/tests/outdir"
            }
            process {
                """
                input[0] = Channel.value(params.tumor)
                input[1] = Channel.value(params.normal)
                input[2] = Channel.fromPath(params.tumorBam)
                input[3] = Channel.fromPath(params.normalBam)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            
            // check expected files
            with(process.out) {
                assert cobalt_ratio_tsv.size() == 1
                assert cobalt_ratio_pcf.size() == 1
            }
        }

    }

}
