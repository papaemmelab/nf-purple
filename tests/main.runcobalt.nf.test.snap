{
    "Should run Cobalt without failures": {
        "content": [
            {
                "0": [
                    "TEST_TUMOR.cobalt.ratio.tsv.gz:md5,f235d46690097a000fdf7ac31bb53108"
                ],
                "1": [
                    "TEST_TUMOR.cobalt.ratio.pcf:md5,7fa4142d0d102e1f3b633d61d154d9c0"
                ],
                "2": [
                    "TEST_NORMAL.cobalt.ratio.pcf:md5,4f035e0ee8e4290d64d24152ea5387b8"
                ],
                "cobalt_normal_ratio_pcf": [
                    "TEST_NORMAL.cobalt.ratio.pcf:md5,4f035e0ee8e4290d64d24152ea5387b8"
                ],
                "cobalt_ratio_pcf": [
                    "TEST_TUMOR.cobalt.ratio.pcf:md5,7fa4142d0d102e1f3b633d61d154d9c0"
                ],
                "cobalt_ratio_tsv": [
                    "TEST_TUMOR.cobalt.ratio.tsv.gz:md5,f235d46690097a000fdf7ac31bb53108"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:25:25.678659"
    }
}